### PR TITLE
self contained example for ask and tell

### PR DESCRIPTION
**Pull request by teytaud**
+self contained
+enough budget for writing something significant

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/test_doc.py [hunk 1]
```diff
@@ -45,15 +45,19 @@ def square(x, y=12):
...
+    optimizer = ng.optimizers.OnePlusOne(parametrization=instrum, budget=100, num_workers=1)
 
     for _ in range(optimizer.budget):
         x = optimizer.ask()
         loss = square(*x.args, **x.kwargs)
         optimizer.tell(x, loss)
 
     recommendation = optimizer.provide_recommendation()
-    # DOC_BASE_4
+    print(recommendation.value)    # DOC_BASE_4
```
Comment (jrapin) [inline nevergrad/optimization/test_doc.py]: put `# DOC_BASE_4` on another line

File: nevergrad/optimization/test_doc.py [hunk 2]
```diff
@@ -45,15 +45,19 @@ def square(x, y=12):
     with futures.ThreadPoolExecutor(max_workers=optimizer.num_workers) as executor:
         recommendation = optimizer.minimize(square, executor=executor, batch_mode=False)
     # DOC_BASE_3
-    optimizer = ng.optimizers.OnePlusOne(parametrization=instrum, budget=10, num_workers=1)
+    import nevergrad as ng
+    def square(x, y=12):
+        return sum((x - .5)**2) + abs(y)
+    instrum = ng.p.Instrumentation(ng.p.Array(shape=(2,)), y=ng.p.Scalar())  # We are working on R^2.
```
Comment (jrapin) [inline nevergrad/optimization/test_doc.py]: the comment is wrong (probably to remove somewhere else then), this is R2xR

File: nevergrad/optimization/test_doc.py [hunk 3]
```diff
@@ -45,15 +45,19 @@ def square(x, y=12):
     with futures.ThreadPoolExecutor(max_workers=optimizer.num_workers) as executor:
         recommendation = optimizer.minimize(square, executor=executor, batch_mode=False)
     # DOC_BASE_3
-    optimizer = ng.optimizers.OnePlusOne(parametrization=instrum, budget=10, num_workers=1)
+    import nevergrad as ng
```
Comment (jrapin) [inline nevergrad/optimization/test_doc.py]: please add skipped lines before and after the function definition, for readibility

Comment (teytaud) [reply nevergrad/optimization/test_doc.py]: Done!

File: nevergrad/optimization/test_doc.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/optimization/test_doc.py]: Done, thx.

File: nevergrad/optimization/test_doc.py (hunk 2 above)
Comment (teytaud) [reply nevergrad/optimization/test_doc.py]: Done, thx.

File: nevergrad/optimization/test_doc.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/optimization/test_doc.py]: Done, thx.

**Comment by teytaud** [comment]
@brockho

**Pull request merged**